flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ResolvedSchema.java

```diff
@@ -163,6 +165,18 @@ public int getColumnCount() {
         return Optional.ofNullable(primaryKey);
     }
 
+    /** Returns the primary key indexes, if any, otherwise returns an empty list. * */
+    public int[] getPrimaryKeyIndexes() {
+        return getPrimaryKey()
+                .map(UniqueConstraint::getColumns)
+                .map(
+                        l -> {
+                            List<String> columns = getColumnNames();
+                            return l.stream().mapToInt(columns::indexOf).toArray();
+                        })
+                .orElseGet(() -> new int[] {});
```

Comment:
The column names are unnecessarily calculated on every iteration.

```suggestion
        final List<String> columns = getColumnNames();
        return getPrimaryKey()
                .map(UniqueConstraint::getColumns)
                .map(pkColumns -> pkColumns.stream().mapToInt(columns::indexOf).toArray())
                .orElseGet(() -> new int[] {});
```